### ISSUE TITLE
unable to build the firmware

### ISSUE THREAD

**Issue by Jeff-WuYo** [ouser555/ergodashv99v5#1]
Hi, 
I've tried to compile my own firmware with latest source code from [ergodash-20230409](https://github.com/ouser555/ergodashv99v5/tree/main/firmware/ergodash-20230409), and I've placed the folder into qmk_firmware/keyboards, and named it "ergodashv99v5". 
When I try to compile using `qmk compile -kb ergodashv99v5/rev2 -km v99v5`, I get the following errer.
```
Ψ Compiling keymap with make --jobs=1 ergodashv99v5/rev2:v99v5


QMK Firmware 0.14.18
☒ ergodashv99v5/rev2: RGB_DI_PIN in config.h is no longer a valid option and should be replaced with WS2812_DI_PIN or APA102_DI_PIN
⚠ ergodashv99v5/rev2: Layout "LAYOUT_4key" is offset on X axis by 0.5
⚠ ergodashv99v5/rev2: Layout "LAYOUT_4key_2u_inner" is offset on X axis by 0.5
⚠ ergodashv99v5/rev2: Layout "LAYOUT_3key_2us" is offset on X axis by 0.5
⚠ ergodashv99v5/rev2: Layout "LAYOUT_3key_1us" is offset on X axis by 0.5
⚠ ergodashv99v5/rev2: Layout "LAYOUT_2key" is offset on X axis by 0.5
Making ergodashv99v5/rev2 with keymap v99v5

☒ ergodashv99v5/rev2: RGB_DI_PIN in config.h is no longer a valid option and should be replaced with WS2812_DI_PIN or APA102_DI_PIN
⚠ ergodashv99v5/rev2: Layout "LAYOUT_4key" is offset on X axis by 0.5
⚠ ergodashv99v5/rev2: Layout "LAYOUT_4key_2u_inner" is offset on X axis by 0.5
⚠ ergodashv99v5/rev2: Layout "LAYOUT_3key_2us" is offset on X axis by 0.5
⚠ ergodashv99v5/rev2: Layout "LAYOUT_3key_1us" is offset on X axis by 0.5
⚠ ergodashv99v5/rev2: Layout "LAYOUT_2key" is offset on X axis by 0.5
avr-gcc (GCC) 12.2.0
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Compiling: platforms/avr/drivers/ws2812_bitbang.c                                                  In file included from /usr/avr/include/avr/io.h:99,
                 from /usr/avr/include/avr/interrupt.h:38,
                 from platforms/avr/drivers/ws2812_bitbang.c:23:
platforms/avr/drivers/ws2812_bitbang.c: In function 'ws2812_setleds':
platforms/avr/drivers/ws2812_bitbang.c:41:18: error: 'WS2812_DI_PIN' undeclared (first use in this function); did you mean 'WS2812_TIMING'?
   41 |     DDRx_ADDRESS(WS2812_DI_PIN) |= pinmask(WS2812_DI_PIN);
      |                  ^~~~~~~~~~~~~
platforms/avr/_pin_defs.h:64:25: note: in expansion of macro '_PIN_ADDRESS'
   64 | #define DDRx_ADDRESS(p) _PIN_ADDRESS(p, 1)
      |                         ^~~~~~~~~~~~
platforms/avr/drivers/ws2812_bitbang.c:41:5: note: in expansion of macro 'DDRx_ADDRESS'
   41 |     DDRx_ADDRESS(WS2812_DI_PIN) |= pinmask(WS2812_DI_PIN);
      |     ^~~~~~~~~~~~
platforms/avr/drivers/ws2812_bitbang.c:41:18: note: each undeclared identifier is reported only once for each function it appears in
   41 |     DDRx_ADDRESS(WS2812_DI_PIN) |= pinmask(WS2812_DI_PIN);
      |                  ^~~~~~~~~~~~~
platforms/avr/_pin_defs.h:64:25: note: in expansion of macro '_PIN_ADDRESS'
   64 | #define DDRx_ADDRESS(p) _PIN_ADDRESS(p, 1)
      |                         ^~~~~~~~~~~~
platforms/avr/drivers/ws2812_bitbang.c:41:5: note: in expansion of macro 'DDRx_ADDRESS'
   41 |     DDRx_ADDRESS(WS2812_DI_PIN) |= pinmask(WS2812_DI_PIN);
      |     ^~~~~~~~~~~~
platforms/avr/drivers/ws2812_bitbang.c: In function 'ws2812_sendarray_mask':
platforms/avr/drivers/ws2812_bitbang.c:168:69: error: 'WS2812_DI_PIN' undeclared (first use in this function); did you mean 'WS2812_TIMING'?
  168 |                      : "r"(curbyte), "I"(_SFR_IO_ADDR(PORTx_ADDRESS(WS2812_DI_PIN))), "r"(maskhi), "r"(masklo));
      |                                                                     ^~~~~~~~~~~~~
platforms/avr/_pin_defs.h:66:26: note: in expansion of macro '_PIN_ADDRESS'
   66 | #define PORTx_ADDRESS(p) _PIN_ADDRESS(p, 2)
      |                          ^~~~~~~~~~~~
platforms/avr/drivers/ws2812_bitbang.c:168:55: note: in expansion of macro 'PORTx_ADDRESS'
  168 |                      : "r"(curbyte), "I"(_SFR_IO_ADDR(PORTx_ADDRESS(WS2812_DI_PIN))), "r"(maskhi), "r"(masklo));
      |                                                       ^~~~~~~~~~~~~
 [ERRORS]
 |
 |
 |
make[1]: *** [builddefs/common_rules.mk:361: .build/obj_ergodashv99v5_rev2_v99v5/ws2812_bitbang.o] Error 1
Make finished with errors
make: *** [Makefile:392: ergodashv99v5/rev2:v99v5] Error 1
```

When I'm at qmk_firmware/keyboards directory, I run `make ergodashv99v5/rev2:v99v5` and get another error.
```
make: *** No rule to make target 'ergodashv99v5/rev2:v99v5'.  Stop.
```

Is the problem relate to qmk? or I mess something up? 

Thanks in advance.

**Comment by ouser555** [comment]
This program is made using QMK 0.20.4 version, please check your QMK version.

**Comment by Jeff-WuYo** [comment]
Thanks for the reply.
I've fix the code, and created a poll request #2 . Now it's able to compile with latest qmk version.
